staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3046,6 +3046,17 @@ type PodSpec struct {
 	// +listMapKey=topologyKey
 	// +listMapKey=whenUnsatisfiable
 	TopologySpreadConstraints []TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty" patchStrategy:"merge" patchMergeKey:"topologyKey" protobuf:"bytes,33,opt,name=topologySpreadConstraints"`
+	// TTLSecondsAfterFinished limits the lifetime of a Pod that has finished
+	// execution (either Complete or Failed). If this field is set,
+	// TTLSecondsAfterFinished after the Job finishes, it is eligible to be
+	// automatically deleted. When the Job is being deleted, its lifecycle
+	// guarantees (e.g. finalizers) will be honored. If this field is unset,
+	// the Job won't be automatically deleted. If this field is set to zero,
+	// the Job becomes eligible to be deleted immediately after it finishes.
+	// This field is alpha-level and is only honored by servers that enable the
+	// TTLAfterFinished feature.
```

Comment:
this comment is correct, but the `pkg/api/pod/util.go#dropDisabledFields` method needs to actually implement dropping this field when the TTLAfterFinished feature is disabled. See the existing feature and use checks in that method for an example, and the existing unit tests in util_test.go for how to test that